vector/src/main/java/im/vector/app/core/platform/VectorBaseActivity.kt

```diff
@@ -363,6 +373,7 @@ abstract class VectorBaseActivity<VB : ViewBinding> : AppCompatActivity(), Maver
 
     override fun onPause() {
         super.onPause()
+        screenEvent?.send(analyticsTracker, analyticsScreenName)
```

Comment:
do we need to be able to override the screen name in the `send`, can screens dynamically change at runtime?

if not we could use a `val` instead of `var` and drop the extra `screenName` parameter 

```kotlin
// or non null and abstract to force all screens to supply a name
protected open val name: ScreenName? = null  

screenEvent = screenName?.let { ScreenEvent(it) }
...

screenEvent?.send(analyticsTracker)
```

Reply:
Yes, the screen name can change during the screen life. I made this for [this particular case - LoginActivity](https://github.com/vector-im/element-android/pull/4734/files#diff-4a938f906dc61c1b687841b80edc339b631e4b1a695c425d9f2ee458ee268149R207)
This is not ideal, I admit and a bit not clear...